fiftyone/core/odm/document.py

```diff
@@ -523,164 +524,108 @@ class Document(BaseDocument, mongoengine.Document):
     meta = {"abstract": True}
 
     def save(self, validate=True, clean=True, **kwargs):
-        """Save the :class:`Document` to the database.
+        """Saves the document to the database.
 
         If the document already exists, it will be updated, otherwise it will
         be created.
 
         Args:
-            validate (True): validates the document
-            clean (True): call the document's clean method; requires
-                ``validate`` to be True
+            validate (True): whether to validate the document
+            clean (True): whether to call the document's ``clean()`` method.
+                Only applicable when ``validate`` is True
 
         Returns:
             self
         """
+        self._save(deferred=False, validate=validate, clean=clean, **kwargs)
+        return self
+
+    def _save(self, deferred=False, validate=True, clean=True, **kwargs):
```

Comment:
Went ahead and merged the deferred vs non-deffered logic into a single function to avoid duplication